client/src/components/graph.jsx

```diff
@@ -0,0 +1,207 @@
+    labelStart += day
+  }
+  const labelEnd = Math.floor(maxX / day) * day
+
+  for (let label = labelStart; label <= labelEnd; label += day) {
+    labels.push({
+      label: new Date(label).toLocaleDateString(),
+      x: timeToX({ minX, maxX, time: label })
+    })
+  }
+  return labels
+}
+
+export default withStyles({
+  root: {
+    marginBottom: '20px',
+    '& .frame__body': {
+      padding: '20px'
+    }
+  },
+  tooltip: {
+    position: 'absolute',
+    pointerEvents: 'none',
+    background: 'var(--cirrus-fg)',
+    color: 'var(--cirrus-bg)',
+    padding: '5px 10px',
+    borderRadius: '5px',
+    margin: '5px'
+  }
+}, class extends Component {
+  state = {
+    division: null,
+    polylines: [],
+    labels: [],
+    tooltipX: 0,
+    tooltipY: 0,
+    tooltipContent: ''
+  }
+
+  componentDidMount() {
+    this.handleFetchData()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.division !== prevProps.division) {
+      this.handleFetchData()
+    }
+  }
+
+  handleFetchData = async () => {
+    const data = await getGraph({
+      division: this.props.division === 'all' ? undefined : this.props.division
+    })
+    if (data.length === 0) {
+      return
+    }
+    let maxX = 0
+    let minX = Infinity
+    let maxY = 0
+    data.graph.forEach((user) => {
+      user.points.forEach((point) => {
+        if (point.time > maxX) {
+          maxX = point.time
+        }
+        if (point.time < minX) {
+          minX = point.time
+        }
+        if (point.score > maxY) {
+          maxY = point.score
+        }
+      })
+    })
+    const labels = getXLabels({ minX, maxX })
+    const polylines = data.graph.map((user) => pointsToPolyline({
+      points: user.points,
+      id: user.id,
+      name: user.name,
+      currentScore: user.points[0].score,
+      maxX,
+      minX,
+      maxY
+    }))
+    this.setState({
+      polylines,
+      labels
+    })
+  }
+
+  handleTooltipIn = (content) => () => {
+    this.setState({
+      tooltipContent: content,
+    })
+  }
+
+  handleTooltipMove = (evt) => {
+    this.setState({
+      tooltipX: evt.clientX,
+      tooltipY: evt.clientY
+    })
+  }
+
+  handleTooltipOut = () => {
+    this.setState({
+      tooltipContent: ''
+    })
+  }
+
+  render({ classes }, { polylines, labels, tooltipContent, tooltipX, tooltipY, tooltipFlipX }) {
+    return (
+      <div class={`frame ${classes.root}`}>
+        <div class='frame__body'>
+          <svg viewBox={viewBox}>
+            <Fragment>
+              {polylines.map(({ points, color, name, currentScore }, i) => (
+                <polyline
+                  key={i}
+                  stroke-linecap='round'
+                  stroke={color}
+                  stroke-width={stroke}
+                  fill='transparent'
+                  points={points}
+                  pointer-events='stroke'
+                  onMouseOver={this.handleTooltipIn(`${name} - ${currentScore} points`)}
```

Comment:
For performance, it would be best not to create a new handler function every render